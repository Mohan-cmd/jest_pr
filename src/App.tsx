import { useEffect, useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import handleOtherMethod from './helper'
import User from './User'

function App() {
  const [count, setCount] = useState(0)
  const [data,setData] = useState('')
  const [reqData,setReqData] = useState('')
  const login =false;
  const [isData,setIsData] = useState(false)
  const [isbtnVisible,setIsbtnVisible] = useState(false)
  const [lastName,setLastName] = useState('')

  useEffect(()=>{
    setTimeout(()=>{
      setIsData(true)
    },1000)
  },[])
  return (
    <>
      <div>
        <a href="https://vite.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" title ='React logo' alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
      <input type="text" placeholder="Enter your name" aria-label='Username'
      value={data} onChange={(e)=>setData(e.target.value)}/>
      <label htmlFor='Password'>Password</label>
      <input type="text" placeholder="Enter your password" id='Password'
        />
      <button onClick={()=>setReqData('updated Data')} name='Submit'>Submit</button> 
      <h1>{reqData}</h1>
      <button onClick={handleOtherMethod}>Other Method</button>
      <button>Click Me1</button>
      <button>Click Me2</button>

      <select>
        <option value='1'>1</option>
        <option value='2'>2</option>
        <option value='3'>3</option>
        <option value='4'>4</option>
        <option value='5'>5</option>
      </select>

      <label htmlFor='skill'>Select</label>
            <input type='checkbox' id='skill' checked={true}/>
      
      <div data-testid="div-test-id">
        testing with testid
      </div>
      <input type='text' defaultValue={'first name'} aria-label='name'/>
       {/* can be tested with getByDisplayValue */}
      {/* <textarea name='textarea' id='textarea' /> */}

      <span title="click me"></span>
      <input
      type='text'
      defaultValue={'mohan fsdv'}
      name='inputttusername'
      id='inputttusername'
      >
      </input>

      <button className='btn-test-class' id='btn-test-id'>Clickee</button>
      <div >Hello World</div>

      login ? <div>Login</div> : <div>Logout</div>
      {isData && <div>Data is loaded</div>}
      <div> afscdf111
         <p>HI</p>
         <p>How</p>
         <h1>Are you</h1>
      </div>

      <button onClick={()=>setIsbtnVisible(!isbtnVisible)}>
        IsButtonVisible
      </button>
      {isbtnVisible && <h1>Buttonisvisible</h1>}

      <input type='text' placeholder='Enter your lasttname' onChange={(e)=>setLastName(e.target.value)} aria-label='lastname'/>
      <h1>{lastName}</h1>
      <User name='dewfd'/>
      
    </>
  )
}

export default App
