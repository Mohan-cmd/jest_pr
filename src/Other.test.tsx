import {render, screen, fireEvent} from "@testing-library/react"
import '@testing-library/jest-dom';
import App from "./App";

test("Other First react app test case",()=>{
    render(<App/>)
    const text = screen.getByText('Vite + React')
    const image = screen.getByTitle('React logo')
    expect(image).toBeInTheDocument();
    expect(text).toBeInTheDocument();
})

test("Other increments count when button is clicked", () => {
    render(<App />);
    const button = screen.getByRole('button', { name: /count is/i });
    expect(button).toHaveTextContent('count is 0');
    fireEvent.click(button);
    expect(button).toHaveTextContent('count is 1');
});

test("Other wverify the input field",()=>{
    render(<App/>);
     const input = screen.getByPlaceholderText('Enter your name');
   // const input = screen.getByRole('textbox')
    
    expect(input).toBeInTheDocument();
    fireEvent.change(input,{target:{value:'mrthygf'}})
    expect(input).toHaveValue('mrthygf')
    expect(input).toHaveAttribute('name','Username')
});

test("snapshot for APp component",()=>{
    const {container} = render(<App/>)
    expect(container).toMatchSnapshot();
})
