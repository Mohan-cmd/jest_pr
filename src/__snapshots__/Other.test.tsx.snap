// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`snapshot for APp component 1`] = `
<div>
  <div>
    <a
      href="https://vite.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/test-file-stub.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/test-file-stub.svg"
        title="React logo"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
  <input
    name="Username"
    placeholder="Enter your name"
    type="text"
    value=""
  />
  <input
    name="Password"
    placeholder="Enter your password"
    type="text"
  />
  <button
    name="Submit"
  >
    Submit
  </button>
  <h1 />
</div>
`;
